src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,151 +201,160 @@ public boolean canBeRotated() {
         return true;
     }
 
-    public long getPower() {
-        return container.getStoredPower();
+
+    public boolean canWork() {
+        if (machineTier == MachineTier.TIER_0) {
+            return fuelRemaining > 0;
+        } else {
+            return container.getStoredPower() >= powerUsage;
+        }
     }
 
+    public boolean cancrush(ItemStack item) {
+        return PulverizerRegistry.containsInput(item);
+    }
+
+    public void burnTime(int i) {
+        if (fuelRemaining == 0) {
+            if (inventory.getStackInSlot(4).getItem() == lastFuelType && cancrush(inventory.getStackInSlot(i))) {
+                fuelRemaining = lastFuelValue;
+
+
+            } else if (inventory.getStackInSlot(4).getItem() != lastFuelType && cancrush(inventory.getStackInSlot(i))) {
+                fuelRemaining = net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4));
+                lastFuelType = inventory.getStackInSlot(4).getItem();
+                lastFuelValue = fuelRemaining;
+            }
+            fuelTotal = fuelRemaining;
+            inventory.decrStackSize(4, 1);
+            this.markDirty();
+            this.markForUpdate();
+        }
+    }
 
     @Override
     public void update() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "update" is 41 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)